nbcelltests/tests/test_test.py

```diff
@@ -60,28 +60,111 @@ def _import_from_path(path, module_name):
     return mod
 
 
+def _generate_test_module(notebook, module_name):
+    """
+    Generate test file from notebook, then import it with __name__ set
+    to module_name, and return it.
+    """
+    tf = tempfile.NamedTemporaryFile(mode='w', suffix='.py', delete=False, encoding='utf8')
+    tf_name = tf.name
+    try:
+        # the module name (__name__) doesn't really matter, but
+        # will be nbcelltests.tests.test_tests.X, where X is
+        # whatever concrete subclass is this method belongs to.
+        generated_module = _import_from_path(run(notebook, filename=tf_name), module_name)
+    finally:
+        try:
+            tf.close()
+        except Exception:
+            pass
+        os.remove(tf_name)
+
+    return generated_module
+
+
 class _TestCellTests(unittest.TestCase):
     # abstract :)
 
     @classmethod
     def setUpClass(cls):
-        """
-        Generate test file from notebook, then import it, and make the
-        resulting module available as "generated_tests" class attribute.
-        """
-        tf = tempfile.NamedTemporaryFile(mode='w', suffix='.py', delete=False, encoding='utf8')
-        tf_name = tf.name
+        assert hasattr(cls, "NBNAME"), "Subclasses must have NBNAME attribute."  # TODO: make actually abstract
+        cls.generated_tests = _generate_test_module(notebook=cls.NBNAME, module_name="nbcelltests.tests.%s.%s" % (__name__, cls.__name__))
+
+    def _assert_skipped(self, mthd, reason):
+        # TODO: actually calling the skipped method here ends
+        # everything! It's some pytest issue, I think. So for now we
+        # are checking that it would be skipped, rather than that it is
+        # definitely skipped.
```

Comment:
We should probably make an issue to track this if we merge as-is.

Reply:
Will make an issue to track this if this PR is merged with this in. 

It could be time consuming to track down (I couldn't track it down in the 5 minutes I tried...); I also feel like we might end up with an overall better way of doing things that obviates this.

Reply:
#117